src/sentry/integrations/bitbucket/webhook.py

```diff
@@ -74,18 +114,11 @@ def event_type(self) -> IntegrationWebhookEventType:
     def __call__(self, event: Mapping[str, Any], **kwargs) -> None:
         authors = {}
 
+        if not (repo := kwargs.get("repo")):
+            raise ValueError("Missing repo")
         if not (organization := kwargs.get("organization")):
             raise ValueError("Missing organization")
 
-        try:
-            repo = Repository.objects.get(
-                organization_id=organization.id,
-                provider=PROVIDER_NAME,
-                external_id=str(event["repository"]["uuid"]),
-            )
-        except Repository.DoesNotExist:
-            raise Http404()
```

Comment:
Since we are now doing signature check on the upper level (in `post()` method), and `repository` is always in the payload (https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Repository), we will always lookup the repo on the upper level as well. Hence passing the repo as kwarg and removing this code.